src/extension/sldService/src/main/java/org/geoserver/sldservice/rest/ClassifierController.java

```diff
@@ -347,231 +346,272 @@ private JSONObject jsonRule(Object obj) {
         return jsonObj;
     }
 
+    private List<Rule> getRasterRules(
+            String property,
+            String method,
+            Integer intervals,
+            Integer intervalsForUnique,
+            boolean open,
+            String customClasses,
+            Boolean reverse,
+            Boolean normalize,
+            CoverageInfo coverageInfo,
+            ColorRamp ramp,
+            boolean continuous)
+            throws Exception {
+        RasterSymbolizerBuilder builder = new RasterSymbolizerBuilder();
+
+        // grab the raster, for the time being, read fully trying to force deferred loading where
+        // possible
+        GridCoverageReader reader = coverageInfo.getGridCoverageReader(null, null);
+        ParameterValue<Boolean> useImageRead = AbstractGridFormat.USE_JAI_IMAGEREAD.createValue();
+        useImageRead.setValue(true);
+        GridCoverage coverage = reader.read(new GeneralParameterValue[] {useImageRead});
+        RenderedImage image = coverage.getRenderedImage();
+        int selectedBand = getSelectedBand(property, image);
+        if (selectedBand != NO_BAND_SELECTED) {
+            ImageWorker iw = new ImageWorker(image);
+            iw.retainBands(selectedBand);
+            image = iw.getRenderedImage();
+        }
+        ColorMap colorMap;
+        try {
+            if (customClasses.isEmpty()) {
+                if ("equalInterval".equals(method)) {
+                    colorMap =
+                            builder.equalIntervalClassification(image, intervals, open, continuous);
+                } else if ("uniqueInterval".equals(method)) {
+                    colorMap = builder.uniqueIntervalClassification(image, intervalsForUnique);
+                } else if ("quantile".equals(method) || "equalArea".equals(method)) {
```

Comment:
So... equalArea for rasters is identical to quantile classification?

Reply:
Indeed it is... one pixel, one value, one unit of area, so counting repeated values and their area becomes the same.

Reply:
wonderful